### DISCUSSION TITLE
Why is neovim unable to locate the packer module?

### DISCUSSION THREAD

**Discussion by wjyoumans** [wbthomason/packer.nvim#741] [Q&A]
I'm trying to install [packer.nvim](https://github.com/wbthomason/packer.nvim) using the quickstart guide. I cloned the repository using 
```
git clone --depth 1 https://github.com/wbthomason/packer.nvim\
 ~/.local/share/nvim/site/pack/packer/start/packer.nvim
```
and created the file `~/.config/nvim/lua/plugins.lua` with these contents:
```
return require('packer').startup(function()
  -- Packer can manage itself
  use 'wbthomason/packer.nvim'
end)
```
My `~/.config/nvim/init.vim` has only the line `lua require('plugins')`.

When I run `nvim` I get the following error:
```
Error detected while processing /home/user/.config/nvim/init.vim:
line    1:
E5105: Error while calling lua chunk: /home/user/.config/nvim/lua/plugins.lua:6: module 'packer' not found:
        no field package.preload['packer']
        no file '/home/user/.config/nvim/lua/packer.lua'
        no file '/home/user/.config/nvim/lua/packer/init.lua'
        no file '/etc/xdg/xdg-i3/nvim/lua/packer.lua'
        ... (many more missing files)
```

I tried checking my `packpath` in neovim with `:set packpath?` but I'm not sure whether it's correct or not.
```
packpath=~/.config/nvim,/etc/xdg/xdg-i3/nvim,/etc/xdg/nvim,~/.local/share/nvim/site,/usr/share/i3/nvi
m/site,/usr/local/share/nvim/site,/usr/share/nvim/site,/var/lib/snapd/desktop/nvim/site,/usr/share/nvim
/runtime,/var/lib/snapd/desktop/nvim/site/after,/usr/share/nvim/site/after,/usr/local/share/nvim/site/a
fter,/usr/share/i3/nvim/site/after,~/.local/share/nvim/site/after,/etc/xdg/nvim/after,/etc/xdg/xdg-i3/n
vim/after,~/.config/nvim/after
```

What am I doing wrong?

**Comment by VesperQuartz** [comment]
I am facing the same issue too

**Comment by goacmola** [comment]
just run it from PowerShell if you on windows

**Comment by yogeshelke** [comment]
I was facing same issue....
followed this :  https://www.youtube.com/watch?v=5rH_JL3qil0

Error is gone for me

**Comment by yi-ye-zhi-qiu** [comment]
Was having the same issue. Just had to upgrade my neovim to 0.5+. I did not realize I was on 0.4. 

The in-depth steps I used to solve this problem:

For me make sure your neovim --version => 0.5+, 
if on Ubuntu first remove neovim 
`sudo apt-get remove neovim`
then reinstall
`sudo apt-get install neovim`
confirm version 
`nvim--version` (should be v0.5+) 

If this condition is satisfied, install Packer using the normal commands for unix/linux: 
```
git clone --depth 1 https://github.com/wbthomason/packer.nvim\
 ~/.local/share/nvim/site/pack/packer/start/packer.nvim
```

You can then configure lua based neovim(0.5+) by, in `~/.config/nvim/` directory folder, creating the below file: 
`~/.config/nvim/init.lua`
Inside this file you can add some stuff like: 

```lua
-- Change tab to be 2 spaces not 8
vim.bo.expandtab = true
vim.bo.shiftwidth = 2
vim.bo.softtabstop = 2

-- Packer
require('packer').startup(function()
  use 'wbthomason/packer.nvim'
end)
```

You may verify the installation by checking that `PackerCompile` and `PackerSync` both work

**Reply by KJunya** [reply]
> confirm version
> `nvim--version` (should be v0.5+)

Had the same issue with Debian 10. I installed by `apt install neovim` which installed version 0.3.
I downloaded the compiled tar.gz of v0.5.1 and it worked. (the latest, at the time being, v0.7.2 did not run as it required higher gcc library or something).

**Reply by nat-418** [reply]
You are not going to successfully run such old Neovim versions with popular plugins. You should consider using Flatpak or Nix to install more recent software on Debian oldstable.

**Comment by tim-hilt** [comment]
@wjyoumans so you didn't find a solution yet? I just tried to open nvim and encountered the same issue. Didn't find a solution yet.

**Reply by mykytak** [reply]
Had same issue, solution was to restart nvim

**Comment by J-Kappes** [comment]
Same issue with NVIM v0.7.2.
I tried changing the vim.init line to `lua require ('packer')` in case it was a typo, and that actually lets nvim start without the error message :S

**Comment by nat-418** [comment]
@wjyoumans your Packer config is missing the `use` parameter in the initial `startup` callback.

Here is the correct code:

```lua
return require('packer').startup(function(use)
  -- Packer can manage itself
  use 'wbthomason/packer.nvim'
end)
```
